src/mame/drivers/8080bw.cpp

```diff
@@ -1318,31 +1318,80 @@ void _8080bw_state::cosmicmo(machine_config &config)
-	PORT_DIPSETTING(    0x00, "2500" )
-	/* SW1:5,6,7 Default ON */
 	PORT_DIPNAME( 0x80, 0x80, "Laser Bonus Info" )      PORT_DIPLOCATION("SW1:1") /* Default is OFF (Info on) */
-	PORT_DIPSETTING(    0x00, DEF_STR( Off ) )
-	PORT_DIPSETTING(    0x80, DEF_STR( On ) )
+	PORT_DIPSETTING(    0x00, DEF_STR( On ) )
+	PORT_DIPSETTING(    0x80, DEF_STR( Off ) )
+  PORT_DIPNAME( 0x08, 0x00, DEF_STR( Bonus_Life ) )   PORT_DIPLOCATION("SW1:2") /* Default is ON (2500) */
+  PORT_DIPSETTING(    0x08, "1500" )
+  PORT_DIPSETTING(    0x00, "2500" )
+  PORT_DIPNAME( 0x03, 0x03, DEF_STR( Lives ) )        PORT_DIPLOCATION("SW1:4,3") /* Default is OFF, OFF (6) */
+  PORT_DIPSETTING(    0x00, "3" )
+  PORT_DIPSETTING(    0x01, "4" )
+  PORT_DIPSETTING(    0x02, "5" )
+  PORT_DIPSETTING(    0x03, "6" )
+
+  PORT_MODIFY("IN0")
+  PORT_BIT( 0x70, IP_ACTIVE_HIGH, IPT_CUSTOM ) PORT_CUSTOM_MEMBER(mw8080bw_state, invaders_in1_control_r)  
+  /* SW1:5-8 Unused but mapped to port 0 in hw. */
+  PORT_DIPUNUSED_DIPLOC( 0x80, 0x00, "SW1:5" )
+  PORT_DIPUNUSED_DIPLOC( 0x07, 0x00, "SW1:6,7,8" )
+
+  /* Port 3 
+      Bit 1 is checked one-shot, at start, in the first seven bytes of code. If set to low the code continues normally to the init code, if set to high the CPU is halted. 
+            I don't know if this is some sort of protection. From PCB, I traced it to pin 12B on edge connector. That pin on the original SIDAM Invasion manual is marked as SPARE.
+      Bit 2 Coin counter protection: If coin counter is disconnected the game stops as described in the original SIDAM user manual:
+            "4. Protezione contatore: il microprocessore controlla che il contatore sia inserito e funzioni regolarmente. Nel caso il contatore venga staccato il
+             programma si arresta e non riprende finchè il contatore non viene regolar-mente ricollegato."
+            In hardware this value is driven by a transistor connected between pin 16A and pin 22A of the edge connector.
+            In software it is handled by the code at address $0079
+  */     
+  PORT_START("IN3")
+  PORT_BIT( 0x02, IP_ACTIVE_HIGH, IPT_OTHER) PORT_NAME("Spare") // Pin 12B of edge connector
+  PORT_BIT( 0x04, IP_ACTIVE_LOW, IPT_OTHER) PORT_NAME("Coin counter protection")
+  PORT_BIT( 0xF9, IP_ACTIVE_LOW, IPT_UNUSED)
 INPUT_PORTS_END
 
 
+/* same as regular invaders without hw shifter */
+void _8080bw_state::invasion(machine_config &config)
+{
+  mw8080bw_root(config);
+
+  /* basic machine hardware */
+  m_maincpu->set_addrmap(AS_IO, &_8080bw_state::invasion_io_map);
+
+  /* 60 Hz signal clocks two LS161. Ripple carry will */
+  /* reset circuit, if LS161 not cleared before.      */
+  WATCHDOG_TIMER(config, m_watchdog).set_vblank_count("screen", 255);
+
+  /* video hardware */
+  m_screen->set_screen_update(FUNC(_8080bw_state::screen_update_invaders));
+
+  /* That's probably wrong. Sidam Invasion uses a totally redesigned hardware layout. To be done later... */
+  /* sound hardware */
```

Comment:
Better to use C++ line comments for short comments – less visual noise.

Reply:
Ok, I prefer C++ convention too but I've used standard C convention because I've get the other states method already defined in the same source file as template (_8080bw_state::invadpt2, spcewars, escmars, ...).
I'll change them...